kinoml/features/dunbrack_cluster.py

```diff
@@ -0,0 +1,177 @@
+    centroid[(4, 'f_phi')]=   -85.0
+    centroid[(4, 'f_psi')]=   145.0
+    centroid[(4, 'f_chi1')]=   49.0
+
+    centroid[(5, 'x_phi')]=  -106.0
+    centroid[(5, 'x_psi')]=   157.0
+    centroid[(5, 'd_phi')]=    69.0
+    centroid[(5, 'd_psi')]=    21.0
+    centroid[(5, 'f_phi')]=   -62.0
+    centroid[(5, 'f_psi')]=   134.0
+    centroid[(5, 'f_chi1')]= -145.0
+
+    assignment = list()
+    for i in range(len(distances)):
+        ## reproduce the Dunbrack clustering
+        ## level1: define the DFG positions
+        if distances[i][0] <= 11.0 and distances[i][1] <= 11.0: # angstroms
+            ## can only be BABtrans
+            assignment.append(7)
+        elif distances[i][0] > 11.0 and distances[i][1] < 14.0:
+            ## can only be BBAminus
+            assignment.append(6)
+        else:
+            ## belong to DFGin and possibly clusters 0 - 5
+            mindist=10000.0
+            cluster_assign = 0
+ 
+            for c in range(6):
+                total_dist = float((2.0 * (1.0-cos((dihedrals[i][0] - centroid[(c, 'x_phi')])*np.pi / 180.0)))
+                + (2.0 * (1.0-cos((dihedrals[i][1] - centroid[(c, 'x_psi')])*np.pi / 180.0)))
+                + (2.0 * (1.0-cos((dihedrals[i][2] - centroid[(c, 'd_phi')])*np.pi / 180.0)))
+                + (2.0 * (1.0-cos((dihedrals[i][3] - centroid[(c, 'd_psi')])*np.pi / 180.0)))
+                + (2.0 * (1.0-cos((dihedrals[i][4] - centroid[(c, 'f_phi')])*np.pi / 180.0)))
+                + (2.0 * (1.0-cos((dihedrals[i][5] - centroid[(c, 'f_psi')])*np.pi / 180.0)))
+                + (2.0 * (1.0-cos((dihedrals[i][6] - centroid[(c, 'f_chi1')])*np.pi / 180.0)))) / 7
+                if total_dist < mindist:
+                    mindist = total_dist
+                    clust_assign = c
+            assignment.append(clust_assign)
+    return assignment 
+
+
+class PDBDunbrack:
+
+    _PDB_DUNBRACK_LIBRARY = Path(f"{user_cache_dir()}/pdb_dunbrack_library.csv")
+
+    def __init__(self):
+        self.pdb_dunbrack_library = self.update()
+
+    def __repr__(self):
+        return f"<PDBDunbrack library located at {self._PDB_DUNBRACK_LIBRARY} contains {self.pdb_dunbrack_library.shape[0]} structures.>"
+
+    def update(self, reinitialize: bool = False) -> pd.DataFrame:
+        """
+        Update DataFrame holding information about kinases from the KLIFS database and the corresponding Dunbrack
+        cluster.
+        Parameters
+        ----------
+        reinitialize: bool
+            If the DataFrame should be built from scratch.
+        Returns
+        -------
+        pdb_dunbrack_library: pd.DataFrame
+            DataFrame holding information about kinases from KLIFS and the corresponding Dunbrack cluster.
+        """
+        from .klifs import query_klifs_database
+        import klifs_utils
+        import MDAnalysis as mda
+        from .protein_struct_features import key_klifs_residues, compute_simple_protein_features
+        from tqdm import tqdm
```

Comment:
Use `tqdm.auto` for nicely behaved output in notebooks and CLI:

```suggestion
        from tqdm.auto import tqdm
```